File: frontend/src/microservices/operations/overall/objects/ManagerBatchDecisionPlannerRequest.ts
// 本文件定义 `BatchConfirmManagerTasksPlanner` 和 `BatchRejectManagerTasksPlanner` 的请求对象。

export type ManagerBatchDecisionPlannerRequest = {
  managerId: string
  managerType: string
  orderItemIds: string[]
  reason?: string | null
  note?: string | null
}

export const managerBatchDecisionPlannerRequestFromJson = (json: string): ManagerBatchDecisionPlannerRequest =>
  JSON.parse(json) as ManagerBatchDecisionPlannerRequest

export const managerBatchDecisionPlannerRequestToJson = (value: ManagerBatchDecisionPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/overall/objects/ManagerBatchDecisionPlannerRequest.scala
// 本文件定义 `BatchConfirmManagerTasksPlanner` 和 `BatchRejectManagerTasksPlanner` 对应的请求对象。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerBatchDecisionPlannerRequest(managerId: String, managerType: String, orderItemIds: List[String], reason: Option[String], note: Option[String])
object ManagerBatchDecisionPlannerRequest:
  given sourceEncoder: Encoder[ManagerBatchDecisionPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[ManagerBatchDecisionPlannerRequest] = deriveDecoder
